sound/soc/sof/pcm.c

```diff
@@ -687,6 +687,37 @@ static int sof_pcm_dai_link_fixup(struct snd_soc_pcm_runtime *rtd,
 	return 0;
 }
 
+static void sof_set_longname(struct snd_soc_component *component)
+{
+	struct snd_sof_dev *sdev = snd_soc_component_get_drvdata(component);
+	struct snd_sof_pdata *pdata = sdev->pdata;
+	char flavour[80];
+	int pos = 0;
+	int i;
+
+	pos += sprintf(flavour, "%s", "SOF");
+	pos += sprintf(flavour + pos, "-%02d%02d%02d-",
+		       sdev->fw_version.major,
+		       sdev->fw_version.minor,
+		       sdev->fw_version.micro);
```

Comment:
@juimonen The deployment model seems tricky. If we add FW version to flavour, one would in theory have to update UCM each time FW version is updated (even micro version), and/or have logic to ignore the version. But in latter case, where do we put this logic of parsing, alsa-lib ucm parser?

Reply:
@kv2019i yes agreed, versioning the fw and api seems overkill. Somehow I see still that topology-file and ucm should be somehow coupled, so that you can update both. This is really assuming that we have this use case that topology can be updated so that ucm has also new features.